lib/rules/newline-before-return.js

```diff
@@ -103,35 +103,81 @@ module.exports = {
         }
 
         /**
-         * Checks whether node is preceded by a newline
-         * @param {ASTNode} node - node to check
-         * @returns {boolean} Whether or not the node is preceded by a newline
+         * Returns the line number of the token before the node that is passed in as an argument
+         * @param {ASTNode} node - The node to use as the start of the calculation
+         * @returns {number} Line number of the token before `node`
          * @private
          */
-        function hasNewlineBefore(node) {
-            const tokenBefore = sourceCode.getTokenBefore(node),
-                lineNumNode = node.loc.start.line;
+        function getLineNumberOfTokenBefore(node) {
+            const tokenBefore = sourceCode.getTokenBefore(node);
             let lineNumTokenBefore;
 
             /**
-             * Global return (at the beginning of a script) is a special case.
-             * If there is no token before `return`, then we expect no line
-             * break before the return. Comments are allowed to occupy lines
-             * before the global return, just no blank lines.
-             * Setting lineNumTokenBefore to zero in that case results in the
-             * desired behavior.
-             */
+            * Global return (at the beginning of a script) is a special case.
```

Comment:
The spaces are preferred in the little-known [ESLint style guide](http://eslint.org/docs/developer-guide/code-conventions#multi-line-comments).